.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -0,0 +1,15 @@
+closes #???
+
+The following tasks have been completed:
+
+ * [ ] Confirmed there are no ReSpec errors/warnings.
+ * [ ] Modified Web platform tests (link)
+ * [ ] Modified MDN Docs (link)
```

Comment:
Is it better to update MDN when there’s at least one conforming implementation or already when the spec change lands? I’d leave this up to the editor to decide. Good checkpoint though to keep in mind.

Reply:
yes, this is the same as the above... generally, don't merge things into the spec until you have one implementation, and fully tested.

Reply:
Has anyone tried this working mode? MDN changes can't be developed in PRs, and I'm not sure what the MDN team thinks about *when* new APIs are best added. When it's available in a stable browser must be OK, but this would be when the feature is about to jump on the train to stable, and there's some risk it might be disabled before stable.

@chrisdavidmills can probably advise.

Reply:
@foolip we use this definiteion of experimental: https://developer.mozilla.org/en-US/docs/MDN/Contribute/Guidelines/Conventions_definitions#Experimental

Generally we like to have at least a working implementation of a feature before we will consider documenting it, even of that is behind a flag/pref, although we'd prefer to also have a spec that seems fairly stable, and we'd prefer it to be available in release at least somewhere, to give more weight to the idea that it will stick around.

When we are asked to document something super early, we have to make a jugement call as to whether it is worth it. When a spec change lands is fine, as long as there is evidence that it is being actively implemented, but when it jumps onto the train is better.

Reply:
so yeah, the underlying assumption is that an implementation lands before the PR is merged - and there is a commitment from another implementer to also work on feature. Alternative, a request is made from to the MDN team to document the feature (in Gecko, we have a "dev-doc-needed" flag for this).

Reply:
I don't _think_ there's a chicken-and-egg problem here but can CONTRIBUTING.md be updated at the same time to provide guidance for what order contributors should work on each of these items? There are a lot of implicit dependencies not captured here.

Reply:
I agree - and I'm sorry for that. The underlying assumptions are:

1. people are implicitly operating using something similar to http://whatwg.org/working-mode
1. In Firefox, at least (and no doubt the same at Google), we require things like "Intent to Implement", then when sending patches they undergo the normal code review processes, require tests that are hopefully web platform tests, and when a feature lands we document it on MDN.

So, the assumption is that the spec and implementation are happening in parallel: The Editor(s) are not necessarily implementing: but they are gathering feedback from actual *real* implementations in a browser engine.